### PR TITLE
Feat: Avatar group to allow square overflow

### PR DESCRIPTION
**Pull request by RiteshMakan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

1. Related issue or discussion links [here](https://github.com/ant-design/ant-design/issues/36691).

### 💡 Background and solution

When making all Avatar Square in Avatar.Group, the overflow component renders as a circle and cannot be changed to a square as needed by providing the shape property to the Group.

In the screenshot below we can see that the ant-avatar ant-avatar-circle classnames are being applied to the overflow component.

The end user experience would be to maintain the consistency of the shape of Avatars across the .Group component.

The solution was to provide an additional prop to the `Group` to accept `maxShape` prop as either `'circle' | 'square'` the default will be kept `circle` to avoid breaking previous implimentations.

### 📝 Changelog

This is a PR to resolve the following issue:
Ant Avatar.Group overflow item seems to be hardcoded to a circle shape and cannot be changed to square

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by RiteshMakan** [comment]
@afc163 or someone, can you please help me. I am not sure what to do from here after submitting the PR

File: components/avatar/group.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface GroupProps {
   prefixCls?: string;
   maxCount?: number;
   maxStyle?: React.CSSProperties;
+  maxShape?: 'circle' | 'square';
```
Comment (MadCcc) [inline components/avatar/group.tsx:18]:
```suggestion
  shape?: 'circle' | 'square';
```

File: components/avatar/index.en-US.md [hunk 2]
```diff
@@ -34,4 +34,5 @@ Avatars can be used to represent people or objects. It supports images, `Icon`s,
 | maxPopoverPlacement | The placement of excess avatar Popover | `top` \| `bottom` | `top` |  |
 | maxPopoverTrigger | Set the trigger of excess avatar Popover | `hover` \| `focus` \| `click` | `hover` | 4.17.0 |
 | maxStyle | The style of excess avatar style | CSSProperties | - |  |
+| maxShape | The shape of excess avatar container | `circle` \| `square` | `circle` |  |
```
Comment (MadCcc) [inline components/avatar/index.en-US.md:37]:
```suggestion
| shape | The shape of excess avatar container | `circle` \| `square` | `circle` |  4.23.0  |
```

**Review by MadCcc — COMMENTED** [review]
Run `npm run test -u` to update snapshot.

**Comment by MadCcc** [comment]
And fill the changelog plz.

**Comment by MadCcc** [comment]
Don't merge master, merge feature istead.

**Pull request closed without merging**